src/IO/H5/Python/ExtractDatFromH5.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+from spectre.IO import H5 as spectre_h5
+import spectre.DataStructures
+import numpy as np
+import os
+import shutil
+
+
+def extract_dat_files(filename, **kwargs):
+    # Need both of these because it's way easier to list all dat files with h5py
+    # but we still need spectre_h5 to actually get the data
+    h5file = spectre_h5.H5File(filename, "r")
+
+    all_dat_files = h5file.all_dat_files()
+
+    if kwargs["list"]:
+        print_str = "\n ".join(all_dat_files)
+        print("Dat files within '{}':\n {}".format(filename, print_str))
+        return
+
+    out_dir = kwargs["output_directory"]
+
+    if not os.path.exists(out_dir):
+        os.mkdir(out_dir)
+    else:
+        if kwargs["force"]:
+            shutil.rmtree(out_dir, ignore_errors=True)
+            os.mkdir(out_dir)
+        else:
+            raise ValueError(
+                "Could not make directory '{}'. Already exists.".format(
+                    out_dir))
+
+    for dat_path in all_dat_files:
+        split_path = dat_path.split("/")
+        dat_dir = out_dir + "/".join(split_path[:-1])
+        dat_filename = out_dir + dat_path
+
+        if not os.path.exists(dat_dir):
+            os.makedirs(dat_dir)
+
+        dat_file = h5file.get_dat(dat_path[:-4])
```

Comment:
do you need to remove the extension? I thought our code handled that internally. It probably checks that the extension is `.dat` but other than that I think it'll work

Reply:
Yes. Our code always adds the extension regardless of what is given, so `h5file.get_dat("/path/to/file.dat")` errors with
```
Cannot open the object '/path/to/file.dat.dat' because it does not exist.
```

Reply:
I think that's a bug in the code then. It looks like it should be checking for the extension and only adding it if necessary

Reply:
ah, `File` doesn't allow it but the individual files support it... Sigh...